packages/zoe/src/objArrayConversion.js

```diff
@@ -18,3 +18,50 @@ export const objToArray = (obj, keywords) => {
   );
   return keywords.map(keyword => obj[keyword]);
 };
+
+export const assertSubset = (whole, part) => {
+  part.forEach(key => {
+    assert.typeof(key, 'string');
+    assert(whole.includes(key), details`key ${key} was not an expected key`);
+  });
+};
+
+// Keywords must equal the keys of obj
+export const filterObjToArray = (obj, keywords) => {
+  // ensure all keys are in keywords
+  const keys = Object.getOwnPropertyNames(obj);
+  assertSubset(keywords, keys);
+  // ensure all keywords are defined on obj
+  return keywords.map(keyword => {
+    assert(
+      obj[keyword] !== undefined,
+      details`obj[keyword] must be defined for keyword ${keyword}`,
+    );
+    return obj[keyword];
+  });
+};
+
+// return a new object with only the keys in subsetKeywords
```

Comment:
Should it enforce that all the subsetKeywords be present in obj?  If not, it's worth saying so in the comment.

Reply:
Yes, it should.